app/models/events/tag.rb

```diff
@@ -0,0 +1,45 @@
+class Events::Tag < ApplicationModel
+  attribute :source_type, :string
+  attribute :subject_type, :string
+  attribute :action, :string
+
+  def initialize(value)
+    values = value.split(":")
+
+    source_type = values[0]
+    subject_type = values[1...-1]&.join(":").presence || source_type
+    action = values[-1]
```

Comment:
This is the critical code from Tag, the rest is added to load and dump this class. (beside to_struct which is a pure function)

Without this complexity you add a few methods like Event.subject_type(tag) and then the tag can just be a string.

I think this is worthwhile, but I'm open to other approaches or happy to just let this slide. We could make this change even after it merges because this serializes back to the original string.